buffer_test.go

```diff
@@ -95,3 +96,60 @@ func TestBufferSafety(t *testing.T) {
 	t.Log("post gc 2")
 	verify()
 }
+
+type ByteCounter struct {
```

Comment:
Nit: I don't think we need to export this type, so probably best to have it start lower case.
```suggestion
type byteCounter struct {
```

Reply:
Done.